src/server/connection.cpp

```diff
@@ -79,20 +86,23 @@ void Connection::handle_read(const boost::system::error_code &error, std::size_t
     }
 
     boost::beast::error_code ec;
-    http_request_parser->put(boost::asio::buffer(incoming_data_buffer, bytes_transferred), ec);
+    http_request_parser->put(boost::asio::buffer(incoming_data_buffer), ec);
```

Comment:
We used it wrong for the case when request size > 8192 too... 🤦🏻

Reply:
https://www.boost.org/doc/libs/develop/libs/beast/doc/html/beast/ref/boost__beast__http__parser/put.html

> This function attempts to incrementally parse the HTTP message data stored in the caller provided buffers. Upon success, a positive return value indicates that the parser made forward progress, consuming that number of bytes. In some cases there may be an insufficient number of octets in the input buffer in order to make forward progress. This is indicated by the code [error::need_more](https://www.boost.org/doc/libs/develop/libs/beast/doc/html/beast/ref/boost__beast__http__error.html). When this happens, the caller should place additional bytes into the buffer sequence and call [put](https://www.boost.org/doc/libs/develop/libs/beast/doc/html/beast/ref/boost__beast__http__parser/put.html) again. The error code [error::need_more](https://www.boost.org/doc/libs/develop/libs/beast/doc/html/beast/ref/boost__beast__http__error.html) is special. When this error is returned, a subsequent call to [put](https://www.boost.org/doc/libs/develop/libs/beast/doc/html/beast/ref/boost__beast__http__parser/put.html) may succeed if the buffers have been updated. Otherwise, upon error the parser may not be restarted.